django/contrib/staticfiles/storage.py

```diff
@@ -450,6 +452,82 @@ def stored_name(self, name):
         # the intermediate files on disk may be corrupt; avoid an infinite loop.
         raise ValueError("The name '%s' could not be hashed with %r." % (name, self))
 
+    def prepare_pattern(self, pattern):
```

Comment:
I don't think this is the way to approach this patch
I expect that the regex patterns are updated considering they are a hard-coded list, rather than adding an extra processing step to programmatically update the regex